internal/controller/resource_manager.go

```diff
@@ -101,34 +124,54 @@ func (r *ResourceManager) watchEvents(ctx context.Context, resource string, apiG
 		r.logger.Error("Error setting up watch on resource", zap.Error(err))
 		return err
 	}
-	for event := range watcher.ResultChan() {
-		switch event.Type {
-		case watch.Error:
-			r.logger.Error("Watcher event has returned an error", zap.Error(err))
-			return err
-		case watch.Bookmark:
-			continue
-		default:
-		}
-		convertedData, err := getObjectMetadataFromRuntimeObject(event.Object)
-		if err != nil {
-			r.logger.Error("Cannot convert runtime.Object to metav1.ObjectMeta", zap.Error(err))
-			return err
-		}
-		resource := event.Object.GetObjectKind().GroupVersionKind().Kind
-		metadataObj, err := convertMetaObjectToMetadata(r.logger, ctx, *convertedData, r.clientset, resource)
-		if err != nil {
-			r.logger.Error("Cannot convert object metadata", zap.Error(err))
-			return err
-		}
-		err = streamMutationObjectData(r.streamManager, metadataObj, event.Type)
-		if err != nil {
-			r.logger.Error("Cannot send resource mutation", zap.Error(err))
-			return err
-		}
 
+	var event watch.Event
+	for {
+		select {
+		case <-ctx.Done():
+			r.logger.Debug("Disconnected from CloudSecure",
+				zap.String("reason", "context cancelled"),
+				zap.String("api_group", apiGroup),
+				zap.String("resource", resource),
+			)
```

Comment:
Add these log fields into a new logger as context fields, so that all logs in this methods have them.
Add them at the top of the method:
```
	logger := r.logger.With(
		zap.String("api_group", apiGroup),
		zap.String("resource", resource),
	)
```
Then just use the `logger` local variable in the rest of the method, e.g. here:
```suggestion
			logger.Debug("Disconnected from CloudSecure", zap.Error(ctx.Err())
```